runners/s3-benchrunner-rust/src/lib.rs

```diff
@@ -0,0 +1,87 @@
+use serde::Deserialize;
+use std::{fs::File, io::BufReader, process};
+
+mod transfer_manager;
+pub use transfer_manager::TransferManagerRunner;
+
+fn exit_with_skip_code(msg: &str) -> ! {
+    eprintln!("Skipping benchmark - {msg}");
+    process::exit(123)
+}
+
+/// All configuration for a benchmark runner.
+/// Includes values from workload json file, and from the command line
+#[derive(Debug)]
+pub struct BenchmarkConfig {
+    pub workload: WorkloadConfig,
+    pub bucket: String,
+    pub region: String,
+    pub target_throughput_gigabits: f64,
+}
+
+/// From the workload's JSON file
+#[derive(Debug, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct WorkloadConfig {
+    pub version: i32,
+    pub files_on_disk: bool,
+    pub checksum: Option<String>,
+    pub max_repeat_count: i32,
+    pub max_repeat_secs: i32,
```

Comment:
IMO, if it's meant to represent a duration, just use the `Duration` type.
```suggestion
    pub max_repeat_duration: Duration,
```

Reply:
This struct is the deserialized version of a [Workload JSON file](https://github.com/awslabs/aws-crt-s3-benchmarks/tree/main/workloads#readme).

It doesn't seem worth all the extra code to transform this into a duration type, or having a separate struct where we store these things with more Rust-like names and types.